File: domain/models/workshop.py
from dataclasses import dataclass, field
from typing import Optional


@dataclass
class WorkshopItem:
    pubfileid: str
    title: str = ""
    preview_url: str = ""
    author: str = ""
    author_url: str = ""
    description: str = ""
    file_size: str = ""
    posted_date: str = ""
    updated_date: str = ""
    tags: dict = field(default_factory=dict)
    rating_star_file: str = ""
    num_ratings: str = ""
    is_installed: bool = False
    is_downloading: bool = False


@dataclass
class WorkshopFilters:
    search: str = ""
    sort: str = "trend"
    days: str = "7"
    category: str = ""
    type_tag: str = ""
    age_rating: str = ""
    resolution: str = ""
    misc_tags: list[str] = field(default_factory=list)
    genre_tags: list[str] = field(default_factory=list)
    excluded_misc_tags: list[str] = field(default_factory=list)
    excluded_genre_tags: list[str] = field(default_factory=list)
    asset_type: str = ""
    asset_genre: str = ""
    script_type: str = ""
    required_flags: list[str] = field(default_factory=list)
    page: int = 1


@dataclass
class WorkshopPage:
    items: list[WorkshopItem] = field(default_factory=list)
    current_page: int = 1
    total_pages: int = 1
    total_items: int = 0
    filters: Optional[WorkshopFilters] = None

File: src/types/workshop.ts
export interface CollectionRef {
  id: string;
  title: string;
  item_count: number;
}

export interface WorkshopItem {
  pubfileid: string;
  title: string;
  preview_url: string;
  author: string;
  author_url: string;
  description: string;
  file_size: string;
  posted_date: string;
  updated_date: string;
  tags: unknown;
  rating_star_file: string;
  num_ratings: string;
  is_collection: boolean;
  collections?: CollectionRef[];
}

export interface WorkshopPage {
  items: WorkshopItem[];
  total_pages: number;
  total_items: number;
  current_page: number;
}

export interface InstalledWallpaper {
  pubfileid: string;
  folder: string;
  project_json_path: string;
  has_pkg: boolean;
  title: string;
  preview: string;
  description: string;
  file_type: string;
  tags: string[];
  size_bytes: number;
  installed_ts: number;
}
